docs/gateway.md

```diff
@@ -43,6 +43,18 @@ your query string to explicitly specify the filename. For example:
 
 > https://ipfs.io/ipfs/QmfM2r8seH2GiRaC4esTjeraXEachRt8ZsSeGaWTPLyMoG?filename=hello_world.txt
 
+## IPNS & DNSLink
```

Comment:
This doesn't seem like the right place for this documentation. This document is about gateway features and these are just general IPFS features.

Maybe we should add  sentence to the top saying that the gateway provides access to files/directories served on IPFS and then briefly describe `/ipns/...` and `/ipfs/...`?